### PR TITLE
Added data-path command line argument

### PR DESCRIPTION
**Pull request by lucamaraschi**
**Depends on**: [node-clinic-doctor - PR 165](https://github.com/nearform/node-clinic-doctor/pull/165)

This PR adds the possibility of specifying a user defined path to store the data collected by Doctor.
PRs for bubbleprof and flame needs to be done to extend this functionality to the CLI

### CODE REVIEW

**Comment by mcollina** [comment]
You should add the new option to https://github.com/nearform/node-clinic/blob/master/docs/clinic-doctor.txt as well.

**Comment by BridgeAR** [comment]
Should we not call it `path` instead of `data-path`?

**Comment by lucamaraschi** [comment]
Esthetically I was not really convinced about `data-path` but I thought it was more expressive than `data`. What about `destination`?

**Comment by mcollina** [comment]
How about `--dest`?

**Comment by BridgeAR** [comment]
I am fine with `dest`.

File: docs/clinic-doctor.txt [hunk 1]
```diff
@@ -23,4 +23,4 @@
   --collect-only             Do not process data on terminiation
   --visualize-only datapath  Build or rebuild visualization from data
   --sample-interval interval Sample interval in milliseconds
-  --on-port                  Run a script when the server starts listening on a port.
+  --on-port                  Run a script when the server starts listening on a port.  --dest                     Destination for the collect data (default .).
```
Comment (mcollina) [inline docs/clinic-doctor.txt]: I think this should be on 2 lines.

**Comment by mcollina** [comment]
Can you please add a unit test in https://github.com/nearform/node-clinic/blob/master/test/cli-doctor-full.test.js?

File: bin.js [hunk 2]
```diff
@@ -92,9 +92,11 @@ const result = commist()
         'debug'
       ],
       string: [
+        'dest',
```
Comment (mcollina) [inline bin.js]: this seems to be duplicated.

File: test-local/cli-doctor-collect-only.test.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+const fs = require('fs')
+const path = require('path')
+const test = require('tap').test
+const cli = require('../test/cli.js')
+
+test('clinic doctor --collect-only --dest \'./foo\' - no issues', function (t) {
+  cli({ relayStderr: false }, [
+    'clinic', 'doctor', '--collect-only', '--dest', './foo',
+    '--', 'node', '-e', 'setTimeout(() => {}, 300)'
```
Comment (BridgeAR) [inline test-local/cli-doctor-collect-only.test.js]: Suggestion: we do not have to run the application for at least 300 ms and could use a significantly lower value.

File: test-local/cli-doctor-collect-only.test.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+})
+
+test('clinic doctor --collect-only - bad status code', function (t) {
+  cli({ relayStderr: false }, [
+    'clinic', 'doctor', '--collect-only', '--dest', './foo',
+    '--', 'node', '-e', 'process.exit(1)'
+  ], function (err, stdout, stderr) {
+    t.strictDeepEqual(err, new Error('process exited with exit code 1'))
+    t.strictEqual(stdout, 'To stop data collection press: Ctrl + C\n')
+    t.ok(stderr.includes('process exited with exit code 1'))
```
Comment (BridgeAR) [inline test-local/cli-doctor-collect-only.test.js]: Suggestion also add an assertion that verifies that no clinic folder has been added to the directory due to the failure.

File: test-local/cli-doctor-collect-only.test.js (hunk 3 above)
Comment (lucamaraschi) [reply test-local/cli-doctor-collect-only.test.js]: I agree but I took it for consistency from https://github.com/nearform/node-clinic/blob/master/test-local/cli-flame-collect-only.test.js#L11. Shall we keep them aligned around `100`?

Comment (BridgeAR) [reply test-local/cli-doctor-collect-only.test.js]: I would likely just use a value as low as 25 or so. It does not matter how much we actually profile as long as it's enough for the test to pass.

And yes, the other value should likely also be reduced.

File: test-local/cli-doctor-collect-only.test.js (hunk 4 above)
Comment (lucamaraschi) [reply test-local/cli-doctor-collect-only.test.js]: This is more complicated as the `stderr` does not expose the target folder or any information about the pid.

Comment (BridgeAR) [reply test-local/cli-doctor-collect-only.test.js]: Hm, true. We could just read the directory but it's probably fine as it is.

Comment (BridgeAR) [reply test-local/cli-doctor-collect-only.test.js]: (We could work around this by reading the folders before and after and only compare the length of all folders but it should not be that important).

**Pull request merged**